apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx

```diff
@@ -116,7 +117,7 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
 
   const paymentLink = `/api/integrations/stripepayment/subscription?intentUsername=${
     inputUsernameValue || usernameFromStripe
-  }&action=${usernameChangeCondition}&callbackUrl=${router.asPath}`;
+  }&action=${usernameChangeCondition}&callbackUrl=${WEBAPP_URL}${router.asPath}`;
```

Comment:
Recent changes on next broke callbackUrl to be only path adding full url now should solve the issue.

Reply:
https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object

Reply:
How did you decide to implement within the frontend instead of backend?

Reply:
To be fair I don't remember but having url on server will mean that it can only handle one url, instead of having the UI component in different places and they should be on charge of their url to return, this means if component gets moved around places it will be itself in charge of having to provide return url.